### PR TITLE
Remove legacy material element support.

### PR DESCRIPTION
**Pull request by bernardkwok**
We no longer support down-grading to material elements so get rid of this + any look filters related to this.

### CODE REVIEW

File: source/MaterialXRuntime/RtFileIo.cpp [hunk 1]
```diff
@@ -856,72 +856,6 @@ namespace
         return destNode;
     }
 
-    void writeMaterialElement(NodePtr mxNode, DocumentPtr doc, const RtWriteOptions* options)
```
Comment (bernardkwok) [inline source/MaterialXRuntime/RtFileIo.cpp:859]: Nixing this downgrade path.

File: source/MaterialXRuntime/RtFileIo.cpp [hunk 2]
```diff
@@ -1183,18 +1106,9 @@ namespace
...
-        {
-            writeCollections(stage, *doc, options);
-            writeLooks(stage, *doc, options);
-            writeLookGroups(stage, *doc, options);
-        }
-
-        for (auto & mxNode: materialElements) {
-            writeMaterialElement(mxNode, doc, options);
-        }
+        writeCollections(stage, *doc, options);
```
Comment (bernardkwok) [inline source/MaterialXRuntime/RtFileIo.cpp:1109]: Nix look logic here which should not have been applied.

File: source/MaterialXRuntime/RtFileIo.cpp [hunk 3]
```diff
@@ -1157,13 +1091,6 @@ namespace
             else if (typeName == RtNode::typeName())
             {
                 NodePtr mxNode = writeNode(prim, doc, options);
```
Comment (niklasharrysson) [inline source/MaterialXRuntime/RtFileIo.cpp:1093]: Storing the result to a local variable could be removed here.

**Review by niklasharrysson — APPROVED** [review]
Looks good, and I only had a small remark above.

**Pull request merged**